bot/cogs/dogeboard.py

```diff
@@ -0,0 +1,294 @@
+        )
+        embed.set_thumbnail(url=message.author.avatar_url)
+
+        # Set the image to the first one if there is one
+        for attachment in message.attachments:
+            *_, ext = attachment.filename.split(".")
+            if ext in ["jpg", "png", "jpeg", "gif", "gifv"]:
+                embed.set_image(url=attachment.url)
+                break
+
+        await channel.send(embed=embed)
+        await message.add_reaction("✅")
+
+    @Cog.listener()
+    async def on_raw_reaction_add(self, payload: RawReactionActionEvent) -> None:
+        """Handle event when users add reactions."""
+        # Ensure the caches are ready
+        await self.bot.wait_until_ready()
+
+        dogeboard_data = await self._get_guild_data(payload.guild_id)
+        if not dogeboard_data:
+            return
+
+        if str(payload.emoji) == dogeboard_data.dogeboard_emoji:
+
+            channel = self.bot.get_channel(payload.channel_id)
+            message = await channel.fetch_message(payload.message_id)
+
+            dogeboard_reaction: Optional[Reaction] = None
+            for reaction in message.reactions:
+
+                if reaction.emoji == "✅":
+                    if self.bot.user in await reaction.users().flatten():
+                        # If the bot has set a checkmark reaction to this message,
+                        # it's already been DogeBoarded
+                        return
+
+                elif str(reaction.emoji) == str(payload.emoji):
+                    dogeboard_reaction = reaction
+
+            if dogeboard_reaction:
+                if dogeboard_reaction.count >= dogeboard_data.dogeboard_reactions_required:
+                    dogeboard_channel = self.bot.get_channel(dogeboard_data.dogeboard_id)
+                    await self._send_dogeboard_message(message, dogeboard_channel)
+
+    @group(brief="Collate the best message of the server using reactions")
+    async def dogeboard(self, ctx: Context) -> None:
+        """Command group for DogeBoard."""
+        if not ctx.subcommand_passed:
+            dogeboard_data = await self._get_guild_data(ctx.guild.id)
+            embed = Embed(title="DogeBoard", color=Color.blurple())
+
+            if not dogeboard_data:
+                embed.description = "DogeBoard is not configured, " \
+                                    "use `.dogeboard channel <channel>` to start."
+            else:
+                embed.add_field(name="Emoji", value=dogeboard_data.dogeboard_emoji)
+                embed.add_field(
+                    name="Channel",
+                    value=ctx.guild.get_channel(dogeboard_data.dogeboard_id).mention
+                )
+                embed.add_field(name="Required", value=str(dogeboard_data.dogeboard_reactions_required))
+
+            await ctx.send(embed=embed)
+
+    @dogeboard.command(brief="Set the dogeboard_emoji used for the DogeBoard, Custom Emojis Only")
+    async def emoji(self, ctx: Context) -> None:
```

Comment:
Changing the emoji (by reacting to the bot message with the new emoji) causes the message with the reaction to be sent to the dogeboard. This can easily be negated by excluding any bot message from being starred. I think this would be acceptable and will likely implement this myself really quickly.